### PR TITLE
[Rust] Minor SIMD benchmark fix set minimal CPU target for AVX2

### PR DESCRIPTION
**Pull request by eddyxu**
* Set default CPU target to haswell to enable AVX2 routine. Haswell is the oldest CPU still available on AWS and GCP now.  One generation newer (Broadwell) does not bring statistically significant improvement in our benchmarks.
* Fixed `pprof` build on Linux.
* Update the benchmark to reduce the calls to `Float32Array::values()` to make the comparision fair.

### CODE REVIEW

**Comment by eddyxu** [comment]
Haswell was released in 2013, which is really, really old.

We could probably be a bit more aggressive, tho, like supporting up to 2016/2017 year CPUs tho, which is likely to be mid-ter of AWS/GCP offering, and 2017/2018 MacBooks. 

For example, macOS Ventura [does not support pre-2017 machines](https://support.apple.com/en-us/HT213264)

File: rust/.cargo/config.toml [hunk 1]
```diff
@@ -15,5 +15,8 @@ rustflags = [
     "-Wclippy::use_self",
 ]
 
+[target.x86_64-unknown-linux-gnu]
+rustflags = ["-C", "target-cpu=haswell", "-C", "target-feature=+avx2"]
+
 [target.aarch64-apple-darwin]
-# rustflags = ["-C", "target-cpu=native", "-C", "target-feature=+neon"]
+rustflags = ["-C", "target-cpu=apple-m1", "-C", "target-feature=+neon"]
```
Comment (changhiskhan) [inline rust/.cargo/config.toml:22]: What about m2?

File: rust/benches/l2.rs [hunk 2]
```diff
@@ -39,10 +39,9 @@ fn l2_arrow_arity(x: &Float32Array, y: &Float32Array) -> f32 {
...
 
 #[inline]
-fn l2_auto_vectorization(x: &Float32Array, y: &Float32Array) -> f32 {
-    x.values()
-        .iter()
-        .zip(y.values().iter())
+fn l2_auto_vectorization(x: &[f32], y: &[f32]) -> f32 {
+    x.iter()
+        .zip(y.iter())
         .map(|(a, b)| (a - b).powi(2))
```
Comment (changhiskhan) [inline rust/benches/l2.rs:45]: Is it faster to do a-b times itself instead of power?

File: rust/.cargo/config.toml (hunk 1 above)
Comment (eddyxu) [reply rust/.cargo/config.toml:22]: This flag is for supporting 'any cpu after the target'. Similar to the x86_64 target setting to `haswell` which was a 2013 CPU, whichs that we dont support any CPUs older than that.

File: rust/benches/l2.rs (hunk 2 above)
Comment (eddyxu) [reply rust/benches/l2.rs:45]: https://rust.godbolt.org/z/KG48nn6ec  as you can see from the compiler output, this `powi(2)` is compiled to `vmulss`

```Assembly 
vsubss  xmm1, xmm1, dword ptr [rdx + 4*rsi + 24]
vmulss  xmm1, xmm1, xmm1
vaddss  xmm0, xmm0, xmm1
```

which is the similar as `(a-b) * (a-b)` (https://rust.godbolt.org/z/f67vn6PbW)

However the former has better loop unrolling.

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**